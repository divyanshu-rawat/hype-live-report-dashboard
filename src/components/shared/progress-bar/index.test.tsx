import React from 'react';

import { cleanup, render } from '@testing-library/react';

import { ProgressBar } from '.';

afterEach(cleanup);

test('renders ProgressBar', () => {
    render(<ProgressBar
        percent={0.8}
        type="md"
    />);
});

test('ProgressBar Percentage should equal to 70%', () => {
    const { getByTestId } = render(<ProgressBar
        percent={0.7}
        type="md"
    />);
    expect(getByTestId('ProgressBarPercentage')).toHaveTextContent('70%');
});

it('should take a ProgressBar snapshot', () => {
    const { asFragment } = render(<ProgressBar
        percent={80}
        type="md"
    />);
    expect(asFragment()).toMatchSnapshot();
});
