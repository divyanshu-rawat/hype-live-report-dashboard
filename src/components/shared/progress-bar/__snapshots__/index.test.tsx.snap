// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a ProgressBar snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="css-1685clg"
      width="100%"
    >
      <p
        class="css-1ety6vy"
      >
        Sell through
      </p>
      <span
        data-testid="ProgressBarPercentage"
      >
        8000%
      </span>
    </div>
    <div
      class="css-h932pf"
      type="md"
      width="100%"
    />
  </div>
</DocumentFragment>
`;
